lint_test/avoid_unused_parameters_test.dart

```diff
@@ -10,11 +10,70 @@ import 'package:flutter/material.dart';
 
 /// Check the `avoid_unused_parameters` rule
 
+typedef MaxFun = int Function(int a, int b);
+
+typedef Named = String Function({required String text});
+
+// expect_lint: avoid_unused_parameters
+final MaxFun bad = (int a, int b) => 1;
+
+final MaxFun good = (int a, _) => a;
+
+final MaxFun ok = (int _, int __) => 1;
+
+final MaxFun goodMax = (int a, int b) {
+  return a + b;
+};
+
+// expect_lint: avoid_unused_parameters
+final Named _named = ({required text}) {
+  return '';
+};
```

Comment:
Don't think this is what we want -- no good way to satisfy the lint in case we don't actually need to use a named parameter, which may be coming from a 3rd-party API that we have not control over.

Let's exclude named parameters from the lint for now.